addons/im_livechat/static/src/core/public_web/thread_model_patch.js

```diff
@@ -55,6 +55,9 @@ patch(Thread.prototype, {
         if (this.channel_type !== "livechat" || !this.correspondent) {
             return super.displayName;
         }
+        if (this.custom_channel_name) {
+            return this.custom_channel_name;
+        }
```

Comment:
I would put `super` call when there's `custom_channel_name`, and make base code use `isChatChannel` condition too, to take into account livechat and whatsapp.

Reply:
I would personally go with putting the code in `thread_model` and have livechat and whatsapp work using the base code in mail for `isChatChannel` conversations.

Agree that the `whatsapp` code should not depend on `im_livechat`: both modules have no mutual dependency so they must work with or without the other.

The less override the better, I would prefer keeping no override in `whatsapp` for this and it just works thanks to it being already flagged as a chat channel conversation.